kernel/aster-nix/src/syscall/fsync.rs

```diff
@@ -18,6 +18,22 @@ pub fn sys_fsync(fd: FileDesc) -> Result<SyscallReturn> {
             .ok_or(Error::with_message(Errno::EINVAL, "not inode"))?;
         inode_handle.dentry().clone()
     };
-    dentry.sync()?;
+    dentry.sync(false)?;
```

Comment:
> As well as flushing the file data, fsync() also flushes the metadata information associated with the file.

So fsync should sync the data and metadata. I think you should add another method `sync_data`, which is similar to https://doc.rust-lang.org/std/fs/struct.File.html#method.sync_data. The `sync` method can be renamed to `sync_all`

Reply:
Thanks! I think it is better.